tests/unit/editors/Bugs/tests.html

```diff
@@ -1289,6 +1301,33 @@
 				equal(editor.igPercentEditor("value"), 0.029, "value should be 0.029"); //fails prior fix and returns 29 instead
 					$("#695Editor").remove();
 			});
+
+				testId = 'Bug 809 - Wrong value is set when we have isLimitedToListValues: true and revertIfNotValid: false';
+				test(testId, 3, function () {
+				var editor =  $("#809Editor");
+				input = editor.igNumericEditor("field"),
+				editor.select();
+				keyInteraction(38, input); //Arrow Up
+				input.val("");
+				editor.blur();
+				equal(editor.igNumericEditor("value"), "", "Value should be empty string");
+				editor.igNumericEditor("option","allowNullValue", "true");
+				editor.select();
+				keyInteraction(38, input); //Arrow Up
+				input.val(null);
```

Comment:
This will result in an empty string actually, if you want to test the null case you'll either need a different null value or use the value method instead.